### PR TITLE
docs: Update deprecation OEP with suggestions on release cycle alignment

### PR DESCRIPTION
**Pull request by timmc-edx**

### CODE REVIEW

File: oeps/processes/oep-0021-proc-deprecation.rst
Comment (timmc-edx) [inline oeps/processes/oep-0021-proc-deprecation.rst:189]: Note to self, feedback from Diana: "if we want to require adding deprecations into the release notes, we should put that line in the Removing/Removed section" -- plan on making this change.

**Comment by felipemontoya** [comment]
I like this PR and the sentiment behind it. :+1:

File: oeps/processes/oep-0021-proc-deprecation.rst
Comment (pdpinch) [reply oeps/processes/oep-0021-proc-deprecation.rst:189]: As the release notes writer for the past two named releases, I definitely tried to include deprecation information. For the most part, I picked it up from reading the named release page in the Confluence wiki, and by skimming through DEPR tickets. I'd like to think some details could be extracted from commit messages as well. 

What do you all have in mind?

Comment (timmc-edx) [reply oeps/processes/oep-0021-proc-deprecation.rst]: I did a bit of a rewrite and included a note in the Accepted section. Does this seem about right?

**Review by feanil — APPROVED** [review]

**Comment by idegtiarov** [comment]
I like this PR as well. 👍🏻

File: oeps/processes/oep-0021-proc-deprecation.rst
Comment (sarina) [inline oeps/processes/oep-0021-proc-deprecation.rst:201]: Nit: The removal can actually come as soon as Maple is cut - at that point, all changes will be in Nutmeg.

**Review by sarina — APPROVED** [review]

File: oeps/processes/oep-0021-proc-deprecation.rst
Comment (timmc-edx) [reply oeps/processes/oep-0021-proc-deprecation.rst:201]: This is true, but I'm a little concerned that removals that happen shortly after branch cut may interfere with fixes that need to be backported. Maybe that's too rare to be concerned about, though.

**Comment by pdpinch** [comment]
You can see my draft of the deprecations section of the nutmeg release notes at https://edx--2056.org.readthedocs.build/projects/open-edx-release-notes/en/2056/nutmeg.html#deprecations

If you have comments or questions about the nutmeg release notes specifically, please comment on the PR https://github.com/openedx/edx-documentation/pull/2056

File: oeps/processes/oep-0021-proc-deprecation.rst
Comment (sarina) [reply oeps/processes/oep-0021-proc-deprecation.rst:201]: Yeah. I don't feel entirely strongly either way. I think sometimes it can be really annoying to have to hold up artificially another two months, and practically, people are merging breaking things after a given release is cut. This also has the benefit of more time on master and thus more tested before the next release. But! This is why I said "nit" - I would defer to you.

Comment (sarina) [reply oeps/processes/oep-0021-proc-deprecation.rst:201]: Maybe a compromise could be something like,

```
implement the removal some time after Maple is released (so that the
removal itself will land in Nutmeg). Note that while removals technically
can come as soon as a named release is _cut_, it is discouraged as the
removal may interfere with fixes that need to be backported. Any deployment...
```

**Comment by sarina** [comment]
@timmc-edx this has two approvals and I think is good to go. I added one suggestion to our back and forth from earlier, but I really don't feel super strongly about it. As author and someone who participates in DEPR, I defer to you. Let's get this merged!

File: oeps/processes/oep-0021-proc-deprecation.rst
Comment (timmc-edx) [reply oeps/processes/oep-0021-proc-deprecation.rst]: I updated the language a bit -- more or less what you said. I think it's ready for merge if you do. :-) (Would you be acting as Arbiter here? I'm not totally clear on the OEP-update process, even after having looked at OEP-1.)

Comment (sarina) [reply oeps/processes/oep-0021-proc-deprecation.rst]: I find OEP-1 written like OEPs will hardly ever change because a change implies a new OEP.

My personal interpretation is a reasonably-sized change to an OEP should have an arbiter, but smaller things (ie, clarifying points or updating the OEP to reflect how we actually operate) don't need one - just good review.

To me this seems like it falls in the latter bucket - it's a small change that mostly just clarifies how we commonly operate. If you think it's larger, the "correct" way to do things would be to first ask the original arbiter and if s/he isn't available, find a new one.

Comment (timmc-edx) [reply oeps/processes/oep-0021-proc-deprecation.rst]: Makes sense!

**Comment by sarina** [comment]
Thanks for remembering the changelog! Here is an extra 👍🏻 from me!

**Pull request merged**
